src/ripple/app/ledger/Ledger.cpp

```diff
@@ -1762,7 +1694,9 @@ void Ledger::updateFees ()
     }
 
     {
-        StaticScopedLockType sl (sPendingSaveLock);
+        // VFALCO Why not do this before calling getASNode?
+        std::lock_guard<
+            std::mutex> lock(mutex_);
```

Comment:
`std::lock_guard<decltype(mutex_)> lock(mutex_);` is an interesting idiom if you object to a type alias for the `lock_guard`. I wouldn't change this - I don't think we'll change the mutex type. We can also bug Howard about adding template type inference for constructors to the standard :-)